pkg/cmd/backuprestore/backupserver.go

```diff
@@ -0,0 +1,142 @@
+package backuprestore
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"k8s.io/klog/v2"
+
+	"github.com/adhocore/gronx/pkg/tasker"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	backupVolume = "/var/backup/etcd/"
+	backupFolder = "current-backup"
+)
+
+var shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}
+
+type backupServer struct {
+	schedule  string
+	timeZone  string
+	enabled   bool
+	scheduler *tasker.Tasker
+	backupOptions
+}
+
+func NewBackupServer(ctx context.Context) *cobra.Command {
+	backupSrv := &backupServer{
+		backupOptions: backupOptions{errOut: os.Stderr},
+	}
+
+	cmd := &cobra.Command{
+		Use:   "backup-server",
+		Short: "Backs up a snapshot of etcd database and static pod resources without config",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			klog.Infof("hello from backup server :) ")
+
+			if err := backupSrv.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+			if err := backupSrv.Run(ctx); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+	backupSrv.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (b *backupServer) AddFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&b.enabled, "enabled", false, "enable backup server")
+	fs.StringVar(&b.schedule, "schedule", "", "schedule specifies the cron schedule to run the backup")
+	fs.StringVar(&b.timeZone, "timezone", "", "timezone specifies the timezone of the cron schedule to run the backup")
+	cobra.MarkFlagRequired(fs, "enabled")
+
+	b.backupOptions.AddFlags(fs)
+}
+
+func (b *backupServer) Validate() error {
+	if !b.enabled {
+		klog.Infof("backup-server is disabled")
+		return nil
+	}
+
+	if len(b.schedule) == 0 {
+		err := errors.New("missing required flag: --schedule")
+		klog.Error(err)
+		return err
+	}
+
+	b.backupOptions.backupDir = backupVolume + backupFolder
+	return b.backupOptions.Validate()
+}
+
+func (b *backupServer) Run(ctx context.Context) error {
+	klog.Infof("hello from backup server Run() :) ")
+
+	b.scheduler = tasker.New(tasker.Option{
```

Comment:
If we continue with Tasker, this function could be condensed to just calling `b.scheduler.Run()`. Tasker handles the same signals by default https://github.com/adhocore/gronx/blob/main/pkg/tasker/tasker.go#L294 and can be configured to take a parent context https://github.com/adhocore/gronx/blob/main/pkg/tasker/tasker.go#L113.